File: 2025/R/solve_day1_part2.R
solve_day1_part2 <- function(input) {
  pos <- 50L
  dial_size <- 100L
  zero_counter <- 0L

  dir <- substr(input, 1L, 1L)
  val <- as.integer(substr(input, 2L, nchar(input)))
  turns <- ifelse(dir == "R", val, -val)

  for (turn in turns) {
    for (k in seq_len(abs(turn))) {
      if (turn < 0) {
        dir <- -1L
      } else {
        dir <- 1
      }

      pos <- pos + dir

      if (pos == -1) {
        pos <- 99
      }
      if (pos == 100) {
        pos <- 0
      }

      zero_counter <- zero_counter + (pos == 0)
    }
  }

  zero_counter
}

File: 2025/python/solve_day1_part2.py
def solve_day1_part2(text):
    zero_counter = 0
    dial_size = 100
    pos = 50

    for line in text:
        if line[0] == "R":
            step = 1
        else:
            step = -1

        amount = int(line[1:])

        for _ in range(amount):
            pos = (pos + step) % dial_size
            if pos == 0:
                zero_counter += 1

    return(zero_counter)
